Visitor can view articles by category

psnoya opened the pull request:
[Visitor can view articles by category CLIENT](https://www.pivotaltracker.com/story/show/176053440)
### User Story
```
As a visitor
In order to find an article I want to read
I would like to be able to view articles by category
```
## Changes proposed in this pull request:
* Front-end category component -created with Menu items
* Add index by category function in getArticles module
## What I have learned working on this feature:
* How to work with redux in the menu bar
## Screenshots (Client)
![55555](https://user-images.githubusercontent.com/69510622/104004616-39bb6180-51a4-11eb-90fd-b894501ed542.png)

salindholm committed:
Creating a test for visitor can view articles by category and creating fixture file

KyunginNa committed:
Adding josn in categories_data.json file
Creating CategoryMenu component
Importing CategoryMenu in App component

KyunginNa committed:
Adding route for articlesIndex in test
Adding index_by_category function
Adding onClick event in CategoryMenu component

psnoya committed:
Created a single_article_sports.json
Modifying category menu
Formating
Modifying the category test

ViktorHek commented on cypress/integration/userCanViewArticlesByCategory.feature.js:
```diff
@@ -0,0 +1,54 @@
+describe("A user can,", () => {
+  describe("successfully,", () => {
```
you can delete describe("successfully") if you don't have a sad path

ViktorHek commented on src/components/CategoryMenu.jsx:
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
```
remove useState

psnoya committed:
Fixed the review comments changes

Ehugo2000 commented on src/modules/getArticles.js:
```diff
@@ -2,18 +2,23 @@ import axios from "axios";
   async index(dispatch) {
-    let result = await axios.get("/articles");
-    dispatch({type: "SET_NEWS_FEED", payload: result.data.articles})
+    const result = await axios.get("/articles");
```
Do you need to use a try-catch block here and in the category request?

KyunginNa replied:
It is because we didn't create an error handler in the api

emtalen replied:
But I agree with Erik here, you should have a catch block always, you could get a 500 error for example

KyunginNa replied:
Oh okay! We added try and catch blocks!

emtalen commented on src/modules/getArticles.js:
```diff
@@ -2,18 +2,23 @@ import axios from "axios";
     }
   },
+
+  async index_by_category(categoryId, dispatch) {
```
are you really finding the category by the id in the api?

KyunginNa replied:
We were using label, so we fixed it in the api!

psnoya committed:
Adding specificArticle in the rootReducer
Modyfied app.jsx
Modyfied CategoryMenu
it to it.only in userCanViewArticlesByCategory.feautre.js

salindholm committed:
Fixing categories paths in categorymenu and app, removing it.only in test

salindholm committed:
Removing debugger, adding newsFeed as an empty array in rootReducer

salindholm committed:
Removing categories route from app, changing to-paths in category menu, removing debugger, removing empty newsfeed array in view_article case in rootreducer

salindholm committed:
Refactoring code

Ehugo2000 commented on src/App.jsx:
```diff
@@ -2,18 +2,20 @@ import React from "react";
+      <CategoryMenu />
       <Switch>
         <Route exact path="/" component={DisplayArticlesList}></Route>
         <Route exact path="/articles/:id" component={DisplayArticle}></Route>
```
Why did you remove the category Route?

KyunginNa replied:
It didn't work, so we're still trying to figure it out. Yet, it doesn't need to be in this user story so we deleted it for now.

Ehugo2000 commented on src/components/DisplayArticlesList.jsx:
```diff
@@ -6,7 +6,7 @@ import { useDispatch, useSelector } from "react-redux";
 const DisplayArticlesList = () => {
   const dispatch = useDispatch();
-  const { newsFeed } = useSelector(state => state);
+  const { newsFeed } = useSelector((state) => state);
```
Might be able to clean up code, remove () around ```(state)```.

KyunginNa replied:
Removed it :)

Ehugo2000 reviewed (commented):
Left some comments

KyunginNa committed:
Adding chromeWebSecurity in cypress.json
Formatting code

salindholm committed:
Adding try catch to index and index_by category functions in modules

salindholm committed:
Updating error message to a set payload as a generic error
